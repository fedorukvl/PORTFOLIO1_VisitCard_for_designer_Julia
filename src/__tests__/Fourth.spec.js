import React from "react";
import { shallow } from "enzyme";
import { Fourth } from "../Components/index.js";

describe("Tests of <Third/> component", () => {
	let fourthComponent;
	beforeEach(() => {
		fourthComponent = shallow(<Fourth />);
	});
	afterEach(() => {
		fourthComponent = null;
	});
	describe("first render of <Fourth/> component", () => {
		it("<Fourth/> component renders properly", () => {
			expect(fourthComponent).not.toBeNull();
			expect(fourthComponent.debug()).toMatchSnapshot();
		});
	});

	describe("<Fourth/> component renders elements inside itself", () => {
		it("<Fourth/> component has elements inside", () => {
			expect(fourthComponent.find(".pantsDiv").children()).toHaveLength(
				2
			);
			expect(
				fourthComponent.find(".pantsDiv").last().find("h1").text()
			).toEqual("Выполненный проект");
		});
	});

	describe("checking <Fourth/> component`s props", () => {
		describe("<Third/> wihout props", () => {
			it("without props <Third/> component hasn`t <p> in <div .rightDescriptionDiv>", () => {
				expect(
					fourthComponent.find(".rightDescriptionDiv").find("p")
				).toHaveLength(0);
			});
		});
	});
});
