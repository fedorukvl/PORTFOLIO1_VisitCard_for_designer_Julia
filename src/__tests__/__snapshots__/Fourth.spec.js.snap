// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of <Third/> component first render of <Fourth/> component <Fourth/> component renders properly 1`] = `
"<div className=\\"pantsDiv\\">
  <div className=\\"leftPantsImgDiv\\">
    <img src=\\"10.png\\" alt=\\"pants #2\\" className=\\"pantsProfilePic\\" />
    <img src=\\"9.png\\" alt=\\"pants #1\\" className=\\"pantsPic\\" />
  </div>
  <div className=\\"rightPantsDescriptionDiv\\">
    <h1>
      Выполненный проект
    </h1>
    <br />
  </div>
</div>"
`;
